openlibrary/plugins/worksearch/subjects.py

```diff
@@ -95,6 +95,7 @@ class subjects_json(delegate.page):
 
     @jsonapi
     def GET(self, key):
+        web.header('Content-Type', 'application/json')
```

Comment:
This shouldn't be necessary; that's what the `@jsonapi` decorator does

Reply:
I'll double check, but I added it because the headers didn't get set automatically. I agree the decorator _should_ set the header. I'm not yet convinced it does. I'll run another test locally

Reply:
@cdrini I tested again, and what you say is true for the 200 OK responses, but is not the case for the 400 error message I added, which is annoying, and why I inserted that `web.header` line at the top so it applied to _any_ response generated by this `GET` method. 
Let me know if you can see a nicer way to do this, or return a 400 error that is more integrated with the `@jsonapi` decorator. I was finding it hard to find good examples elsewhere.
API error handling isn't that well implemented in the OL code base, and most of what exists was added by me at various points in the past :) I have not found many good web py examples 
 elsewhere either.

Reply:
Ahhh, looking at the definition of `jsonapi` (which is in infogami!):

https://github.com/internetarchive/infogami/blob/956507c0a8f1eacf0bc633c8c211c97dd8e31496/infogami/plugins/api/code.py#L125-L130

It looks like it's `expect`ing a `infogami.infobase.client.ClientException` (sig: `ClientException(status, message, json)`)

Reply:
Although not sure if that will fix it :/

Reply:
I think editing `jsonapi`; maybe to `except` a `JSONAPIException` might work well? If the `ClientException` doesn't solve the issue.

Reply:
@cdrini I tried using a `ClientException`

```
        if i.limit > MAX_RESULTS:
            msg = json.dumps({'error': 'Specified limit exceeds maximum of %s.' % MAX_RESULTS})
            raise ClientException('400 Bad Request', msg)
```

and the decorator still didn't provide a json content type by itself. I think I'd rather use standard web.py as that is more documented than Infogami.

Reply:
:+1: Seems fair!